calcom/cal.com
# feat: add nextcloud talk app

## What does this PR do?



This PR adds Nextcloud Talk as conferencing app with dynamic link type to the app store. Makes use of Nextcloud Talk Conversations management API
- [Create a conversation](https://nextcloud-talk.readthedocs.io/en/stable/conversation/#creating-a-new-conversation)
- [Delete a conversation](https://nextcloud-talk.readthedocs.io/en/stable/conversation/#delete-a-conversation)

Fixes #9579
/claim #9579



## Mandatory Tasks (DO NOT REMOVE)

- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). N/A

## How should this be tested?



- Enable Nextcloud Talk in Admin Apps and set the following fields
	- `nextcloudTalkHost`
	- `nextcloundTalkPattern` (optional)
	- `nextcloudTalkUser`
	- `nextcloudTalkPassword`
![image](https://github.com/user-attachments/assets/78ef1e9f-e637-4b73-8658-a39e88964f07)
- Install Nextcloud Talk from App Store
![image](https://github.com/user-attachments/assets/7fa53465-97b5-45e7-8180-cb0e2231be35)
![image](https://github.com/user-attachments/assets/5f1e85d5-6e51-4ddf-a3e2-db76f3dd508d)
- Add Nextcloud Talk to an event type
![image](https://github.com/user-attachments/assets/d505c933-6b0c-4df3-82f0-9d1fb09f6301)
- Create event with Nextcloud Talk as location
![image](https://github.com/user-attachments/assets/7cd40175-5be7-459d-b62a-2606bba2965b)
- Verify in Nextcloud Talk that a new (public) conversation has been created
![image](https://github.com/user-attachments/assets/62d28989-f1a3-4fb8-8f5c-c9e1d48f923a)
- Cancel event with Nextcloud Talk location
![image](https://github.com/user-attachments/assets/b634b18d-72e4-4572-8638-4085dc04366a)
- Verify in Nextcloud Talk that the conversation has been removed

## Checklist
